Redundant Redeclaration of "packSize()"

I'm getting lots of diagnostic messages of the form
```
/home/bska/work/opm/src/opm/opm-simulators/opm/simulators/utils/ParallelRestart.hpp:286:15: warning: redundant redeclaration of 'std::size_t Opm::Mpi::packSize(const string&, Dune::FakeMPIHelper::MPICommunicator)' in same scope [-Wredundant-decls]
   std::size_t packSize(const T& data, Dune::MPIHelper::MPICommunicator comm); \
               ^
/home/bska/work/opm/src/opm/opm-simulators/opm/simulators/utils/ParallelRestart.hpp:286:15: note: in definition of macro 'ADD_PACK_PROTOTYPES'
   std::size_t packSize(const T& data, Dune::MPIHelper::MPICommunicator comm); \
               ^~~~~~~~
/home/bska/work/opm/src/opm/opm-simulators/opm/simulators/utils/ParallelRestart.hpp:106:13: note: previous declaration of 'std::size_t Opm::Mpi::packSize(const string&, Dune::FakeMPIHelper::MPICommunicator)'
 std::size_t packSize(const std::string& str, Dune::MPIHelper::MPICommunicator comm);
             ^~~~~~~~
```
both in serial and parallel configurations.  The diagnostics seem to stem from the [explicit](https://github.com/OPM/opm-simulators/blob/202235abe0239276536f6be5f1f38b87c66ed098/opm/simulators/utils/ParallelRestart.hpp#L106) declaration of `packSize()` for `std::string` also being generated from the `ADD_PACK_PROTOTYPES` for [std::string](https://github.com/OPM/opm-simulators/blob/202235abe0239276536f6be5f1f38b87c66ed098/opm/simulators/utils/ParallelRestart.hpp#L302).

Would it be appropriate to remove the explicit declaration or does that declaration also serve some other purpose?

--- a/opm/simulators/utils/ParallelRestart.hpp
+++ b/opm/simulators/utils/ParallelRestart.hpp
@@ -107,8 +107,6 @@ std::size_t packSize(const std::array<T,N>& data, Dune::MPIHelper::MPICommunicat
 
 std::size_t packSize(const char* str, Dune::MPIHelper::MPICommunicator comm);
 
-std::size_t packSize(const std::string& str, Dune::MPIHelper::MPICommunicator comm);
-
 template<class T1, class T2, class C, class A>
 std::size_t packSize(const std::map<T1,T2,C,A>& data, Dune::MPIHelper::MPICommunicator comm);
 

